src/Goerli-DssSpell.t.sol

```diff
@@ -443,23 +443,20 @@ contract DssSpellTest is GoerliDssSpellTestBase {
         scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        if (_count != chainLog.count()) {
-            if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(chainLog.version()))) {
+        if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(chainLog.version()))) {
+            // Fail if the version is not updated and the chainlog count has changed
+            if (_count != chainLog.count()) {
                 emit log_named_string("Error", concat("TestError/chainlog-version-not-updated-count-change-", _version));
                 fail();
             }
-        }
-
-        for(uint256 i = 0; i < chainLog.count(); i++) {
-            (, address _val) = chainLog.get(i);
-            // If the address arrays don't match it's due to a change in the changelog. Fail if the version is not updated.
-            if (_chainlog_addrs[i] != _val) {
-                if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(chainLog.version()))) {
+            // Fail if the chainlog is the same size but local keys don't match the chainlog.
+            for(uint256 i = 0; i < chainLog.count(); i++) {
```

Comment:
```suggestion
            for(uint256 i = 0; i < _count; i++) {
```
Just to be annoying lol

Reply:
Actually I think it would be better to call the whole array as the original one and compare each memory array element against the other. That will reduce drastically the calls to the node.

Reply:
That makes sense. I'll make the change.

Reply:
I don't think we'll see a reduction in calls here since there's a separate loop to populate the local array:

```
for(uint256 i = 0; i < _count; i++) {
            (, address _val) = chainLog.get(i);
            _chainlog_addrs[i] = _val;
        }
```

This would need to be replicated on the other side of the cast.

Reply:
Nvm, I was just mixing stuff with the `list` method which only brings the keys. Then the only one that can be saved in the extra count of the first comment.